### PR TITLE
Machine fingerprint 

### PR DESCRIPTION
**Pull request by davidkelly**
Added ability to get the smbios UUID from windows, similar for Mac.  Not using it for anything yet, just grabbing it.  We have ongoing design efforts which will use this, so this is just an effort to grab the data.

If we have ill-behaved bios, where we don't have a good UUID (or we have a bad mac, or we are on linux, etc...) we generate and save a uuid in settings.  Doesn't survive a reinstall, but is something.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4d0f2c1ebb506ffa45820f6341e1d2ab2ae69349 available at: https://gist.github.com/db36edee7ed32e05bae20dd671f0db93<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9153-4d0f2c1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9153-4d0f2c1.dmg

File: libraries/networking/src/FingerprintUtils.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#ifndef hifi_FingerprintUtils_h
+#define hifi_FingerprintUtils_h
+
+#include <QString>
+
+
+class FingerprintUtils {
```
Comment (davidkelly) [inline libraries/networking/src/FingerprintUtils.h]: yea -- not really a need for non-static members of any sort, as it turned out.

**Comment by hifi-gustavo** [comment]
Build console output for commit: c40aadfea41015548f5cbbb4590c4b8c03c8c842 available at: https://gist.github.com/db36edee7ed32e05bae20dd671f0db93<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9153-c40aadf.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9153-c40aadf.dmg

**Comment by davidkelly** [comment]
# Test Plan
I call this function very early in the startup on interface.  So, if you just start interface, you will see the debug output, with your bios UUID, printed on like the 2nd line of the log.  You can compare that to what you get from the command line: `wmic path win32_computersystemproduct get uuid` on windows.  On mac, look at the output from `ioreg -rd1 -c IOPlatformExpertDevice`.  You will see the IOPlatformUUID in there, and it should match what you see in the logs.

To test the case where you have ill-behaving system that doesn't have the uuid, well you can't.  Unless you are "lucky" enough to have a machine like that.  What I did was make the code return an empty string when trying to get the value, and made sure it created one the first time, set it, and then just used that on subsequent runs.  You can do that if you are inclined.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 296d9b1ddf082acbe20d02cae061627fc44eda65 available at: https://gist.github.com/db36edee7ed32e05bae20dd671f0db93<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9153-296d9b1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9153-296d9b1.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c40aadfea41015548f5cbbb4590c4b8c03c8c842 available at: https://gist.github.com/db36edee7ed32e05bae20dd671f0db93<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9153-c40aadf.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9153-c40aadf.dmg

**Comment by OmegaHeron** [comment]
Testing Windows 10
QA Pass
General smoke test = good.
UUID returned = good (and matches expected value - and is consistent across runs as expected).

Linux -- Ubuntu 16.04.1
QA Pass
Same as above except UUID is manually generated (as expected).

File: libraries/networking/src/FingerprintUtils.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#ifndef hifi_FingerprintUtils_h
+#define hifi_FingerprintUtils_h
+
+#include <QString>
+#include <QUuid>
+
+namespace FingerprintUtils {
```
Comment (jherico) [inline libraries/networking/src/FingerprintUtils.h]: [Coding standard](https://wiki.highfidelity.com/wiki/Coding_Standards#Names_representing_namespaces_should_be_all_lowercase.) dictates that namespaces be all lower case.

**Comment by ZappoMan** [comment]
probably needs to be retested once CR fixes are made.

**Comment by OmegaHeron** [comment]
Will monitor and retest when CR updates completed.

**Comment by hifi-gustavo** [comment]
Build console output for commit: dc47c5fc0c606a75939ee68007f8c4d786412af4 available at: https://gist.github.com/db36edee7ed32e05bae20dd671f0db93<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9153-dc47c5f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9153-dc47c5f.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 562cd12bca0d8efcde4551e96a9a947a2599c80c available at: https://gist.github.com/db36edee7ed32e05bae20dd671f0db93<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9153-562cd12.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9153-562cd12.dmg

**Comment by howard-stearns** [comment]
test windows and mac

**Pull request merged**